app/src/main/java/com/tip/lunchbox/utilities/Validator.java

```diff
@@ -0,0 +1,60 @@
+package com.tip.lunchbox.utilities;
+
+import android.util.Patterns;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Validator {
+
+    public static boolean passwordValidator(String password) {
+        if (password.isEmpty()) {
+            return false;
+        }
+        if (password.length() < 6) {
+            return false;
+        }
+        if (!hasCap(password)) {
+            return false;
+        }
+        if (!hasNum(password)) {
+            return false;
+        }
+        if (!hasSpecial(password)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    public static boolean usernameValidator(String userName) {
+        if (userName.isEmpty()) {
+            return false;
+        }
+        if (userName.startsWith(" ")) {
+            return false;
+        }
+        if (userName.endsWith(" ")) {
+            return false;
+        }
+        return true;
+    }
```

Comment:
You don't really need this.. a `.trim()` on any String is better (in Fragment).. avoids throwing unnecessary errors.

Reply:
No if the username enters a space we shouldn't automatically trim it because he might think that his userid is the one with a space (happens rarely ik but still something to think about)

Reply:
What? 😂